pkg/controller/workloadspread/workloadspread_controller.go

```diff
@@ -426,7 +439,7 @@ func (r *ReconcileWorkloadSpread) getAndUpdateSuitableSubsetName(ws *appsv1alpha
 	for i := range ws.Spec.Subsets {
 		subset := &ws.Spec.Subsets[i]
 		// in case of that this pod was scheduled to the node which matches a subset of workloadSpread
-		matched, preferredScore, err := matchesSubset(pod, node, subset)
+		matched, preferredNodeScore, preferredPodScore, err := matchesSubset(pod, node, subset, subsetPatchFields[subset.Name])
```

Comment:
can we move subsetMissingReplicas checking inside matchesSubset, so that matchesSubset only return whether subset can be matched and corresponding matching score. In such way, all scoring logic is in a single function, much easier to test and understand